### ISSUE TITLE
[Bug] Move color to local resource always wraps in a new ResourceDictionary

### ISSUE THREAD

**Issue by davidortinau** [mfractor/mfractor-feedback#273]
I moved `LinearGradientBrush`s into the `ContentPage.Resources` dictionary and then used refactoring to extract the colors into resources. 

This is what I got (RDs multiplying like rabbits):

<img width="833" alt="Screen Shot 2020-07-18 at 10 41 06 AM" src="https://user-images.githubusercontent.com/41873/87857685-3f6b7500-c8ee-11ea-8848-00b5dc61730a.png">

This is what I expected:

```xml
<ContentPage.Resources>
        <Color x:Key="DarkBlue">#0091FF</Color>
        <Color x:Key="LightBlue">#32C5FF</Color>
        <LinearGradientBrush x:Key="BlueGradient" StartPoint="0,0" EndPoint="1,1">
            <GradientStop Color="{StaticResource DarkBlue}" Offset="0.0"/>
            <GradientStop Color="{StaticResource LightBlue}" Offset="1.0"/>
        </LinearGradientBrush>
        <LinearGradientBrush x:Key="GreenGradient" StartPoint="0,0" EndPoint="1,1">
            <GradientStop Color="#6DD400" Offset="0.0"/>
            <GradientStop Color="#44D7B6" Offset="1.0"/>
        </LinearGradientBrush>
    </ContentPage.Resources>
```

**Comment by matthewrdev** [comment]
@davidortinau Can I confirm if this is occurring ring on Visual Studio Mac or Windows?

**Comment by matthewrdev** [comment]
@davidortinau This was caused by a silly logic bug where the InsertResourceEntry code generator used the file on disk rather than the text buffer to perform the resource insertion. 

This has been fixed and will be available in MFractor 4.2.11.

**Comment by matthewrdev** [comment]
@davidortinau This is available in this build: https://addins.monodevelop.com/Stable/Mac/8.6.6/MFractor.VS.Mac.MFractor-4.2.11.mpack
